catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -115,14 +120,30 @@ private SaxEventToXmlElementConverter startConstructingElement(String uri, Strin
             } else {
                 String attUri = atts.getURI(i);
 
-                if (!checkNamespaceAdded(attUri)) {
+                if (!(checkNamespaceAdded(attUri) && scopeOfNamespacesAdded.peek().get(attUri).contains(scopedNamespaces.get(uri)))) {
                     out.writeNamespace(scopedNamespaces.get(attUri), attUri);
                     addedNamespaces.put(attUri, scopedNamespaces.get(attUri));
                 }
-                out.writeAttribute(scopedNamespaces.get(attUri),
-                        attUri,
-                        atts.getLocalName(i),
-                        atts.getValue(i));
+                try {
+                    out.writeAttribute(scopedNamespaces.get(attUri),
+                            attUri,
+                            atts.getLocalName(i),
+                            atts.getValue(i));
+
+                    /*
+                     * XML doesn't allow for duplicate attributes in an element, e.g.
+                     * no <element attribute=1 attribute=2>
+                     * no <element ns1:attribute=1 ns2:attribute=2 xlmns:ns1=foobar xlmns:ns2=foobar>
+                     * however - if one of the namespaces is the default namespace, this duplication is okay,
+                     * yes <element attribute=1 ns1:attribute=2 xlmns=foobar xmlns:ns1=foobar>
+                     *
+                     * This catch block handles this edge case
+                     */
+                } catch (XMLStreamException e) {
+                    out.writeNamespace("ns1", attUri);
```

Comment:
If I'm reading this correctly, it looks like you're preserving namespace prefixes when you write all the other namespaces out. That means that it's not safe to hardcode this edge case prefix as `ns1` could well already be a declared prefix for another namespace.

Since you're making up a prefix for this edge case anyways, I would also prefer a more explicit prefix than `ns1`. Something like `redeclaredDefaultNS`.